bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/ConnectionValidator.java

```diff
@@ -43,25 +38,18 @@
  */
 public class ConnectionValidator {
 
-    private static ModuleTypeRegistry mtRegistry;
-
-    public static void setRegistry(ModuleTypeRegistry mtRegistry) {
-        ConnectionValidator.mtRegistry = mtRegistry;
-    }
-
     /**
      * The method validates connections between inputs and outputs of modules participated in rule. It compares data
      * types of connected inputs and outputs and throws exception when there is a lack of coincidence.
      *
      * @param r rule which must be checked
      * @throws IllegalArgumentException when validation fails.
      */
-    public static void validateConnections(RuntimeRule r) {
+    public static void validateConnections(ModuleTypeRegistry mtRegistry, RuleImpl r) {
```

Comment:
Shouldn't this work for rules in general, regardless which implementation is used?

Reply:
I agree, will update the pull request.